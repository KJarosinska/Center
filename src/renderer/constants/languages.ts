export const LANGUAGE = {
  default: "en-US",
  defaultShort: "en",
  available: ["en-US"],
}
