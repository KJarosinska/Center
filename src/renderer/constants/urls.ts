export const URL_MAIN = {
  root: "/",
  news: "/news",
  overview: "/overview",
  messages: "/messages",
  phone: "/phone",
  music: "/music",
  calendar: "/calendar",
  tools: "/tools",
  meditation: "/meditation",
  filesManager: "/files-manager",
  tethering: "/tethering",
  settings: "/settings",
  help: "/help",
}

export const URL_TABS = {
  templates: "/templates",
  calls: "/calls",
  dial: "/dial",
  playlist: "/playlist",
  voiceRecorder: "/voice-recorder",
  connection: "/connection",
  notifications: "/notifications",
  audioConversion: "/audio-conversion",
}
