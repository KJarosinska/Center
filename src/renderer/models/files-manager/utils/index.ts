import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
