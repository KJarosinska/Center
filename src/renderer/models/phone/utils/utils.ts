import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
