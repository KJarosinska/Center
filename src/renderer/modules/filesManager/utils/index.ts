import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
