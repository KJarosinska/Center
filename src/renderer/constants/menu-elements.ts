import { defineMessages } from "react-intl"
import { View, views } from "Renderer/constants/views"
import { Type } from "Renderer/components/core/icon/icon.config"
import { MenuGroupTestIds } from "Renderer/components/rest/menu/menu-group-test-ids.enum"

const messages = defineMessages({
  yourPure: { id: "menu.header.yourPure" },
  desktopApp: { id: "menu.header.desktopApp" },
})

const YOUR_PURE_BUTTONS = [
  { button: views.overview, icon: Type.MenuOverview },
  {
    button: views.messages,
    icon: Type.Message,
    testId: MenuGroupTestIds.Messages,
  },
  { button: views.phone, icon: Type.MenuPhone, testId: MenuGroupTestIds.Phone },
  {
    button: views.contacts,
    icon: Type.MenuContacts,
    testId: MenuGroupTestIds.Contacts,
  },
  { button: views.music, icon: Type.MenuMusic },
  { button: views.calendar, icon: Type.Calendar },
  { button: views.tools, icon: Type.MenuTools },
  { button: views.meditation, icon: Type.MenuMeditation },
  { button: views.filesManager, icon: Type.MenuFilesManager },
]

const DESKTOP_APP_BUTTONS = [
  { button: views.tethering, icon: Type.MenuTethering },
  { button: views.settings, icon: Type.MenuSettings },
  { button: views.help, icon: Type.MenuHelp, testId: MenuGroupTestIds.Help },
]

interface Item {
  button: typeof views[View]
  icon: Type
  testId?: MenuGroupTestIds
}

export interface MenuElement {
  items?: Item[]
  label?: {
    id: string
  }
  icons?: Type[]
  connectedPhoneOnly?: boolean
  devModeOnly?: boolean
  openHelpWindow?: () => void
}

export const menuElements: MenuElement[] = [
  {
    items: [{ button: views[View.Onboarding], icon: Type.Send }],
    devModeOnly: true,
  },
  {
    items: [{ button: views[View.News], icon: Type.MenuNews }],
  },
  {
    label: messages.yourPure,
    items: YOUR_PURE_BUTTONS,
    icons: [Type.MenuRange, Type.MenuBattery, Type.Sim, Type.TetheringStatus],
    connectedPhoneOnly: true,
  },
  {
    label: messages.desktopApp,
    items: DESKTOP_APP_BUTTONS,
  },
]
