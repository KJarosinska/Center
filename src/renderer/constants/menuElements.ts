import { defineMessages } from "react-intl"
import { VIEWS } from "Renderer/constants/views"

import MenuCalendar from "Renderer/svg/menu_calendar.svg"
import MenuFilesManager from "Renderer/svg/menu_filesManager.svg"
import MenuHelp from "Renderer/svg/menu_help.svg"
import MenuMeditation from "Renderer/svg/menu_meditation.svg"
import MenuMessages from "Renderer/svg/menu_messages.svg"
import MenuMusic from "Renderer/svg/menu_music.svg"
import MenuNews from "Renderer/svg/menu_news.svg"
import MenuOverview from "Renderer/svg/menu_overview.svg"
import MenuPhone from "Renderer/svg/menu_phone.svg"
import MenuSettings from "Renderer/svg/menu_settings.svg"
import MenuTethering from "Renderer/svg/menu_tethering.svg"
import MenuTools from "Renderer/svg/menu_tools.svg"

import Battery from "Renderer/svg/battery.svg"
import Signal from "Renderer/svg/signal.svg"
import Sim from "Renderer/svg/sim.svg"

const messages = defineMessages({
  yourPure: { id: "menu.header.yourPure" },
  desktopApp: { id: "menu.header.desktopApp" },
})

const YOUR_PURE_BUTTONS = [
  { button: VIEWS.overview, icon: MenuOverview },
  { button: VIEWS.messages, icon: MenuMessages },
  { button: VIEWS.phone, icon: MenuPhone },
  { button: VIEWS.music, icon: MenuMusic },
  { button: VIEWS.calendar, icon: MenuCalendar },
  { button: VIEWS.tools, icon: MenuTools },
  { button: VIEWS.meditation, icon: MenuMeditation },
  { button: VIEWS.filesManager, icon: MenuFilesManager },
]

const DESKTOP_APP_BUTTONS = [
  { button: VIEWS.tethering, icon: MenuTethering },
  { button: VIEWS.settings, icon: MenuSettings },
  { button: VIEWS.help, icon: MenuHelp },
]

export const MENU_ELEMENTS = [
  {
    type: "buttons",
    buttons: [{ button: VIEWS.news, icon: MenuNews }],
  },
  {
    type: "header",
    label: messages.yourPure,
    icons: [Signal, Battery, Sim, MenuTethering],
  },
  {
    type: "buttons",
    buttons: YOUR_PURE_BUTTONS,
  },
  {
    type: "header",
    label: messages.desktopApp,
  },
  {
    type: "buttons",
    buttons: DESKTOP_APP_BUTTONS,
  },
]
