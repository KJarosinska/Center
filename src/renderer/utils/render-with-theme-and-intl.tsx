import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={LANGUAGE.default}
        locale={LANGUAGE.default}
        messages={localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}

export function renderWithTheme(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithTheme<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  const originalConsoleError = console.error
  console.error = (...args: any) => {
    if (
      !(
        args[0].startsWith("[React Intl] Cannot format message:") ||
        args[0].startsWith("[React Intl] Missing message:")
      )
    ) {
      originalConsoleError.call(console, ...args)
    }
  }
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider locale={LANGUAGE.default}>{ui}</IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
