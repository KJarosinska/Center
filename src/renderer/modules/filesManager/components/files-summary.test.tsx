import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { Router } from "react-router"
import { URL_MAIN } from "Renderer/constants/urls"
import FilesSummary from "Renderer/modules/filesManager/components/files-summary.component"
import history from "Renderer/routes/history"
import FilesManager from "Renderer/svg/files-manager.svg"
import Mudita from "Renderer/svg/mudita.svg"
import Music from "Renderer/svg/music.svg"
import VoiceRecorder from "Renderer/svg/voice-recorder.svg"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"

const fakeData = [
  {
    filesType: "Music",
    occupiedMemory: 4294967296,
    filesAmount: 15,
    color: "#6D9BBC",
    icon: Music,
    url: URL_MAIN.music,
  },
  {
    filesType: "Voice Recorder",
    occupiedMemory: 4294967296,
    filesAmount: 3,
    color: "#AEBEC9",
    icon: VoiceRecorder,
    url: "/tools/voice-recorder",
  },
  {
    filesType: "Storage",
    occupiedMemory: 4294967296,
    filesAmount: 85,
    color: "#E3F3FF",
    icon: FilesManager,
  },
  {
    filesType: "Free",
    occupiedMemory: 4294967296,
    color: "#E9E9E9",
    icon: Mudita,
  },
]

test("should match snapshot", () => {
  const { container } = renderWithThemeAndIntl(
    <Router history={history}>
      <FilesSummary memoryChartData={fakeData} />
    </Router>
  )
  expect(container.firstChild).toMatchSnapshot()
})

test("should render", () => {
  const { container } = renderWithThemeAndIntl(
    <Router history={history}>
      <FilesSummary memoryChartData={fakeData} />
    </Router>
  )
  expect(container.firstChild).toBeInTheDocument()
})

test("correct amount of items should render", () => {
  const testId = "files-manager-item"
  const { queryAllByTestId } = renderWithThemeAndIntl(
    <Router history={history}>
      <FilesSummary memoryChartData={fakeData} />
    </Router>
  )
  expect(queryAllByTestId(testId)).toHaveLength(4)
})
