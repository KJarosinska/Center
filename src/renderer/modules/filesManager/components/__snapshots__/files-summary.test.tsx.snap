// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="sc-EHOje hSrDqz"
>
  <div
    class="sc-bwzfXH htgWpD"
    data-testid="files-manager-item"
  >
    <test-file-stub
      classname="sc-bxivhb hSTkeS"
    />
    <div
      class="sc-ifAKCX EtoTg"
    >
      <p
        class="sc-bdVaJa jCnOyj"
      >
        Music
      </p>
      <p
        class="sc-bdVaJa dHRpOS"
      >
        15 files (4.0 GB)
      </p>
    </div>
    <a
      class="sc-htpNat jiaXKm"
      href="#/music"
    >
      <test-file-stub
        image="test-file-stub"
      />
    </a>
  </div>
  <div
    class="sc-bwzfXH gBQHvZ"
    data-testid="files-manager-item"
  >
    <test-file-stub
      classname="sc-bxivhb hSTkeS"
    />
    <div
      class="sc-ifAKCX EtoTg"
    >
      <p
        class="sc-bdVaJa jCnOyj"
      >
        Voice Recorder
      </p>
      <p
        class="sc-bdVaJa dHRpOS"
      >
        3 files (4.0 GB)
      </p>
    </div>
    <a
      class="sc-htpNat jiaXKm"
      href="#/tools/voice-recorder"
    >
      <test-file-stub
        image="test-file-stub"
      />
    </a>
  </div>
  <div
    class="sc-bwzfXH iJhYmU"
    data-testid="files-manager-item"
  >
    <test-file-stub
      classname="sc-bxivhb hSTkeS"
    />
    <div
      class="sc-ifAKCX EtoTg"
    >
      <p
        class="sc-bdVaJa jCnOyj"
      >
        Storage
      </p>
      <p
        class="sc-bdVaJa dHRpOS"
      >
        85 files (4.0 GB)
      </p>
    </div>
  </div>
  <div
    class="sc-bwzfXH iQcxRP"
    data-testid="files-manager-item"
  >
    <test-file-stub
      classname="sc-bxivhb hSTkeS"
    />
    <div
      class="sc-ifAKCX EtoTg"
    >
      <p
        class="sc-bdVaJa jCnOyj"
      >
        Free
      </p>
      <p
        class="sc-bdVaJa dHRpOS"
      >
        4.0 GB
      </p>
    </div>
  </div>
</div>
`;
